### PR TITLE
[FIRRTL] Add ListCreateOp.

### PR DESCRIPTION
**Pull request by dtzSiFive**
Test creation of lists with elements of primitive, aggregate, and object/class types.

Parser support will be added separately.

### CODE REVIEW

**Review by prithayan — APPROVED** [review]
Overall looks good.

File: include/circt/Dialect/FIRRTL/FIRRTLExpressions.td [hunk 1]
```diff
@@ -1136,6 +1136,24 @@ def PathOp : FIRRTLOp<"path", [Pure,
   let assemblyFormat = "$target attr-dict";
 }
 
+def ListCreateOp : FIRRTLOp<"listcreate", [Pure]> {
```
Comment (uenoku) [inline include/circt/Dialect/FIRRTL/FIRRTLExpressions.td:1139]: Can we use `SameTypeOperands` and remove `all_equal` in verification code?

**Review by uenoku — APPROVED** [review]
LGTM

File: include/circt/Dialect/FIRRTL/FIRRTLExpressions.td (hunk 1 above)
Comment (dtzSiFive) [reply include/circt/Dialect/FIRRTL/FIRRTLExpressions.td:1139]: Of course, good call thanks!

**Review by rwy7 — APPROVED** [review]

**Comment by dtzSiFive** [comment]
Thanks all! :+1:

**Pull request merged**